sentinel-core/src/main/java/com/alibaba/csp/sentinel/Tracer.java

```diff
@@ -39,87 +35,81 @@
+     * Trace provided {@link Throwable} to current entry of given entrance context.
      *
      * @param e     exception to record
-     * @param count exception count to add
-     * @since 1.4.2
+     * @param context target entrance context
+     * @since 1.8.0
      */
-    public static void traceContext(Throwable e, int count, Context context) {
+    public static void traceContext(Throwable e, Context context) {
         if (!shouldTrace(e)) {
             return;
         }
 
         if (context == null || context instanceof NullContext) {
             return;
         }
-
-        DefaultNode curNode = (DefaultNode)context.getCurNode();
-        traceExceptionToNode(e, count, context.getCurEntry(), curNode);
-    }
-
-    /**
-     * Trace provided {@link Throwable} and increment exception count to provided entry.
-     *
-     * @param e exception to record
-     * @since 1.4.2
-     */
-    public static void traceEntry(Throwable e, Entry entry) {
-        traceEntry(e, 1, entry);
+        traceEntryInternal(e, context.getCurEntry());
     }
 
     /**
-     * Trace provided {@link Throwable} and add exception count to provided entry.
+     * Trace provided {@link Throwable} and add exception count to current entry in provided context.
      *
      * @param e     exception to record
      * @param count exception count to add
      * @since 1.4.2
      */
-    public static void traceEntry(Throwable e, int count, Entry entry) {
+    @Deprecated
+    public static void traceContext(Throwable e, int count, Context context) {
         if (!shouldTrace(e)) {
             return;
         }
-        if (entry == null || entry.getCurNode() == null) {
+
+        if (context == null || context instanceof NullContext) {
             return;
         }
-
-        DefaultNode curNode = (DefaultNode)entry.getCurNode();
-        traceExceptionToNode(e, count, entry, curNode);
+        traceEntryInternal(e, context.getCurEntry());
     }
 
-    private static void traceExceptionToNode(Throwable t, int count, Entry entry, DefaultNode curNode) {
-        if (curNode == null) {
+    /**
+     * Trace provided {@link Throwable} to the given resource entry.
+     *
+     * @param e exception to record
+     * @since 1.4.2
+     */
+    public static void traceEntry(Throwable e, Entry entry) {
+        if (!shouldTrace(e)) {
             return;
         }
-        for (MetricExtension m : MetricExtensionProvider.getMetricExtensions()) {
-            m.addException(entry.getResourceWrapper().getName(), count, t);
-        }
+        traceEntryInternal(e, entry);
+    }
 
-        // clusterNode can be null when Constants.ON is false.
-        ClusterNode clusterNode = curNode.getClusterNode();
-        if (clusterNode == null) {
+    private static void traceEntryInternal(/*@NeedToTrace*/ Throwable e, Entry entry) {
+        if (entry == null) {
             return;
         }
-        clusterNode.trace(t, count);
+
+        entry.setError(e);
     }
```

Comment:
`clusterNode.trace(t, count);` is removed here, and is it right understanding that the `increaseExceptionQps` now is in the `StatisticSlot#recordCompleteFor` method. But the operation of `cluserNode` seems missing, is it affects?

Reply:
It's included in `recordCompleteFor` method (for both DefaultNode and its ClusterNode)

Reply:
Okay, I found it in `DefaultNode#increaseExceptionQps`.